### DISCUSSION TITLE
Can I use MDX with Fresh?

### DISCUSSION THREAD

**Discussion by o-t-w** [freshframework/fresh#1058] [Q&A]
Fresh feels well-suited to documentation. Is there a guide to using Fresh in conjunction with MDX?

**Comment by lucis** [comment]
Similar case is Deno Docs: https://github.com/denoland/fresh/blob/main/www/routes/docs/%5B...slug%5D.tsx

**Reply by ooker777** [reply]
what should I look at the file?

**Reply by rschristian** [reply]
@ooker777 [Here](https://github.com/denoland/fresh/blob/bd16a65587998482dac8d70bd82e3e210d345563/www/routes/docs/%5B...slug%5D.tsx#L123-L124), primarily.

**Reply by ooker777** [reply]
The code is:
```js
// Parse markdown front matter
const url = new URL(`../../../${entry.file}`, import.meta.url);
const fileContent = await Deno.readTextFile(url);
const { body, attrs } = frontMatter<Record<string, unknown>>(fileContent);
```
I'm still not sure what it does, and how it is related to the question.

**Reply by rschristian** [reply]
It's reading a markdown document as text and parsing it into a body and meta data.

Not quite MDX, but if you just want to write in Markdown, it's pretty close.

**Reply by ooker777** [reply]
I take that it doesn't answer the question?

**Reply by rschristian** [reply]
Depends on your needs.

If you specifically want MDX and only MDX, no, it's not of much help, but if you want Markdown, then it's a great example of how you can implement it.

**Comment by imjamesb** [comment]
Yes, you can not SSR MDX on Deno Deploy though.

**Comment by ooker777** [comment]
2 years have passed. Have you found the answer?

**Comment by CAYdenberg** [comment]
On my [blog](https://livingpixel.io/blog) I use [Markdown directives](https://github.com/remarkjs/remark-directive) to intersperse islands with prose. 

MDX would be slicker, but a little harder to implement.

However, this is part of a cluster of ideas I've been tinkering with regarding using Fresh for "blogging", so if there's interest on your end, please reach out via the contact info on my website.
